demos/audiovisual/module/src/renderaudioroadcomponent.cpp

```diff
@@ -0,0 +1,250 @@
+#include "renderaudioroadcomponent.h"
+
+// External Includes
+#include <entity.h>
+#include <computecomponent.h>
+#include <renderglobals.h>
+#include <planemeshvec4.h>
+#include <mesh.h>
+
+// nap::RenderAudioRoadComponent run time class definition 
+RTTI_BEGIN_CLASS(nap::RenderAudioRoadComponent)
+	RTTI_PROPERTY("AudioRoadComponent",		&nap::RenderAudioRoadComponent::mAudioRoadComponent,	nap::rtti::EPropertyMetaData::Required)
+	RTTI_PROPERTY("Ambient",				&nap::RenderAudioRoadComponent::mAmbient,				nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("Diffuse",				&nap::RenderAudioRoadComponent::mDiffuse,				nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("Specular",				&nap::RenderAudioRoadComponent::mSpecular,				nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("FresnelColor",			&nap::RenderAudioRoadComponent::mFresnelColor,			nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("Fresnel",				&nap::RenderAudioRoadComponent::mFresnel,				nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("Shininess",				&nap::RenderAudioRoadComponent::mShininess,				nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("Alpha",					&nap::RenderAudioRoadComponent::mAlpha,					nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("Highlight",				&nap::RenderAudioRoadComponent::mHighlight,				nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("HighlightLength",		&nap::RenderAudioRoadComponent::mHighlightLength,		nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("Reflection",				&nap::RenderAudioRoadComponent::mReflection,			nap::rtti::EPropertyMetaData::Default)
+	RTTI_PROPERTY("Environment",			&nap::RenderAudioRoadComponent::mEnvironment,			nap::rtti::EPropertyMetaData::Default)
+RTTI_END_CLASS
+
+// nap::RenderAudioRoadComponentInstance run time class definition 
+RTTI_BEGIN_CLASS_NO_DEFAULT_CONSTRUCTOR(nap::RenderAudioRoadComponentInstance)
+	RTTI_CONSTRUCTOR(nap::EntityInstance&, nap::Component&)
+RTTI_END_CLASS
+
+//////////////////////////////////////////////////////////////////////////
+
+
+namespace nap
+{
+	bool RenderAudioRoadComponentInstance::init(utility::ErrorState& errorState)
```

Comment:
Isn't it better to simply create a new render audio road component, instead of deriving from renderable mesh component? Since you override the draw method and require a specific mesh type.

Reply:
Yes, refactored it!